### DISCUSSION TITLE
Superset+Keycloak - JWT - The specified alg value is not allowed

### DISCUSSION THREAD

**Discussion by shkpk** [keycloak/keycloak#25226] [Q&A]
Hello

I have superset integrated with Keycloak, and to make API requests to superset, I'm trying to use JWT token, but I get error
```
{"msg":"The specified alg value is not allowed"}
```

I'm using this to generate token

```
token=$(curl --insecure --request POST http://localhost:8088/realms/test/protocol/openid-connect/token \
--header 'Content-Type: application/x-www-form-urlencoded' \
--data-urlencode "username=test" \
--data-urlencode "password=test123" \
--data-urlencode 'scope=email profile roles' \
--data-urlencode 'grant_type=password' \
--data-urlencode 'client_id=superset' \
--data-urlencode "client_secret=zxxxxxxxxxxxxxxxxxx" | jq -r .access_token)
```

and when I use

```
curl -k -L --request GET http://localhost:8090/api/v1/assets/export/ --header "Authorization: Bearer $token" --header "Content-Type: application/zip"
```
which returns

```
{"msg":"The specified alg value is not allowed"}
```

anyone can help with this?

**Comment by DioCrafts** [comment]
I am having the same issue

**Comment by yeval** [comment]
I am having the same issue. Has anyone found any solution to this?

**Comment by mpgalaxy** [comment]
Hi, I'm using python-keycloak (python 3.12.7) with Keycloak 22 and I ran into the same issue.
For me setting the Default Signature Algorithm in Keycloak to RS256 fixed the problem, had RS512 which didn't work with this client (maybe a bug?)
(Realm settings -> Token -> Default Signature Algorithm)
Maybe this helps someone...

**Comment by DarksChoise** [comment]
Has anyone been able to resolve this? I am experiencing the same error.

**Comment by stianst** [comment]
Don't change the default signature algorithm of Keycloak; change it specifically for the client you use with Superset.

**Reply by DarksChoise** [reply]
How do I do it?

**Reply by stianst** [reply]
Go to `Advanced` and search for `Access token signature algorithm` and change it to something Superset supports.
